### DISCUSSION TITLE
Having trouble getting the kratos-selfservice-ui-node hydra-integration-2021 branch working

### DISCUSSION THREAD

**Discussion by mattbarnicle** [ory/.github#48] [Q&A] [answered]
I cloned the repo, checked out the `hydra-integration-2021` branch, then did the following:

```
cd ./contrib/hydra
sudo docker-compose up --build
```

The `hydra-hydra-1` container prints out this error message and restarts, on an infinite loop:

`
hydra-hydra-1                       | time=2023-02-04T17:30:09Z level=fatal msg=Could not ensure that signing keys for "hydra.openid.id-token" exists. If you are running against a persistent SQL database this is most likely because your "secrets.system" ("SECRETS_SYSTEM" environment variable) is not set or changed. When running with an SQL database backend you need to make sure that the secret is set and stays the same, unless when doing key rotation. This may also happen when you forget to run "hydra migrate sql".. audience=application error=map[message:sqlite select many: sql: Scan error on column index 3, name "pk": converting driver.Value type string ("7e956155-274d-419e-a290-b47ce1aaa28a") to a int: invalid syntax] service_name=ORY Hydra service_version=v1.9.0
`

In the `docker-compose.yml` file, I see this:

```
  hydra:
    image: oryd/hydra:v1.9.0-sqlite
    ...
    environment:
      ...
      - SECRETS_SYSTEM=youReallyNeedToChangeThis
```

I'm still branch new to using Ory and don't understand much about it. Does anyone have a clue what I can investigate to get this working?

**Comment by aeneasr** [comment]
Kratos now natively integrates with Hydra and you no longer need the workaround mentioned here. @vinckr can provide some helpful links if need be

**Comment by mattbarnicle** [comment, marked as answer]
oh, i didn't realize that this implementation was a workaround. yes, I would like more information on how to do this with the latest versions. @vinckr please post that info. thanks.

**Reply by vinckr** [reply]
Hello @mattbarnicle 
check out this thread: https://github.com/ory/kratos/discussions/2976
 
I think this is the best resource available atm. A comprehensive tutorial would be good, any contribution welcome.
